runtime/core/tensor_layout.cpp

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <executorch/runtime/core/exec_aten/exec_aten.h>
+#include <executorch/runtime/core/exec_aten/util/scalar_type_util.h>
+#include <executorch/runtime/core/span.h>
+#include <executorch/runtime/core/tensor_layout.h>
+
+namespace executorch {
+namespace runtime {
+
+namespace {
+Result<size_t> calculate_nbytes(
+    const Span<const int32_t>& sizes,
+    const exec_aten::ScalarType& scalar_type) {
+  ssize_t n = 1;
+  for (ssize_t i = 0; i < sizes.size(); i++) {
+    if (sizes[i] < 0) {
+      return Error::InvalidArgument;
+    }
+    n *= sizes[i];
+  }
+  // Use the full namespace to disambiguate from c10::elementSize.
+  return n * executorch::runtime::elementSize(scalar_type);
+}
+} // namespace
+
+Result<TensorLayout> TensorLayout::create(
+    Span<const int32_t> sizes,
+    Span<const uint8_t> dim_order,
```

Comment:
It'd also be a good idea to validate the dim_order entries with a loop like https://github.com/pytorch/executorch/blob/1bf20e3289da509c4344dd65d685f139fc7a172b/runtime/core/exec_aten/util/dim_order_util.h#L24-L30

And also ensure than sizes.size() == dim_order.size()